tests/test_telemetry_monitor.py

```diff
@@ -0,0 +1,1142 @@
+    "run_command, num_db_nodes, launcher",
+    [
+        pytest.param("auto", 1, "local", id="use auto"),
+        pytest.param("mpirun", 1, "local", id="use mpirun"),
+        pytest.param("srun", 1, "slurm", id="use srun"),
+        pytest.param("srun", 2, "slurm", id="use srun w/2 db"),
+    ],
+)
+def test_multistart_experiment(
+    mlutils: MLUtils,
+    wlmutils: WLMUtils,
+    fileutils: FileUtils,
+    monkeypatch: pytest.MonkeyPatch,
+    run_command: str,
+    num_db_nodes: int,
+    launcher: str,
+):
+    """Run an experiment with multiple start calls to ensure that telemetry is
+    saved correctly for each run"""
+    test_dir = fileutils.make_test_dir()
+
+    if run_command != "auto" and not shutil.which(run_command):
+        assert True, f"{run_command} not supported on test environment"
+        return
+
+    exp_name = "my-exp"
+    exp = Experiment(exp_name, launcher=launcher, exp_path=test_dir)
+    rs_e = exp.create_run_settings(
+        sys.executable, ["printing_model.py"], run_command=run_command
+    )
+    rs_e.set_nodes(1)
+    rs_e.set_tasks(1)
+    ens = exp.create_ensemble(
+        "my-ens",
+        run_settings=rs_e,
+        perm_strategy="all_perm",
+        params={
+            "START": ["spam", "foo"],
+            "MID": ["eggs", "bar"],
+            "END": ["ham", "baz"],
+        },
+    )
+
+    yo_path = fileutils.get_test_conf_path("printing_model.py")
+    ens.attach_generator_files(to_configure=[yo_path])
+
+    rs_m = exp.create_run_settings("echo", ["hello", "world"], run_command=run_command)
+    rs_m.set_nodes(1)
+    rs_m.set_tasks(1)
+    model = exp.create_model("my-model", run_settings=rs_m)
+    model.colocate_db_tcp(port=5757, db_identifier="COLO")
+
+    model_file = mlutils.save_torch_cnn(test_dir, f"model_{uuid.uuid4()}.pt")
+    model.add_ml_model("cnn", "TORCH", model_path=model_file, device="CPU")
+
+    db = exp.create_database(
+        db_nodes=num_db_nodes,  #  if USE_SLURM else 1,
+        port=wlmutils.get_test_port(),
+        interface=wlmutils.get_test_interface(),  # "ipogif0" if _launcher != "local" else "lo",
+        single_cmd=False,
+    )
+
+    exp.generate(db, ens, model, overwrite=True)
+
+    with monkeypatch.context() as ctx:
+        ctx.setattr(cfg.Config, "telemetry_frequency", 1)
+        ctx.setattr(cfg.Config, "telemetry_cooldown", 45)
+
+        exp.start(model, block=False)
+
+        # track PID to see that telmon cooldown avoids restarting process
+        tm_pid = exp._control._telemetry_monitor.pid
+
+        exp.start(db, block=False)
+        assert tm_pid == tm_pid == exp._control._telemetry_monitor.pid
+        try:
+            exp.start(ens, block=True, summary=True)
+            assert tm_pid == tm_pid == exp._control._telemetry_monitor.pid
+        finally:
+            exp.stop(db)
+            assert tm_pid == tm_pid == exp._control._telemetry_monitor.pid
+            time.sleep(3)  # time for telmon to write db stop event
```

Comment:
The reduction in size of ensemble will turn 8 members into 2, which should go a long way to reducing test duration w/only 3 nodes & 2 dbs (leaving only 1 being able to run non-databases). I'd like to push for another node in the test allocation instead of reducing test complexity further... especially when we're considering _adding_ db nodes in comment above

Reply:
was decided to drop the clustered db perm from this test as it was not the focus of what is being tested